grpc/pb-marshalling.go

```diff
@@ -318,22 +359,27 @@ func PBToAuthz(pb *corepb.Authorization) (core.Authorization, error) {
 		}
 		challs[i] = chall
 	}
-	expires := time.Unix(0, pb.ExpiresNS).UTC()
+	var expires *time.Time
+	if pb.ExpiresNS != 0 {
+		c := time.Unix(0, pb.ExpiresNS).UTC()
+		expires = &c
+	}
 	authz := core.Authorization{
 		ID:             pb.Id,
 		Identifier:     identifier.ACMEIdentifier{Type: identifier.DNS, Value: pb.Identifier},
 		RegistrationID: pb.RegistrationID,
 		Status:         core.AcmeStatus(pb.Status),
-		Expires:        &expires,
+		Expires:        expires,
 		Challenges:     challs,
 	}
 	return authz, nil
 }
 
 // orderValid checks that a corepb.Order is valid. In addition to the checks
-// from `newOrderValid` it ensures the order ID and the Created field are not nil.
+// from `newOrderValid` it ensures the order ID and the Created fields are not
+// the zero value.
 func orderValid(order *corepb.Order) bool {
-	return order.Id != 0 && order.CreatedNS != 0 && newOrderValid(order)
+	return order.Id != 0 && order.CreatedNS != 0 && order.Created != nil && newOrderValid(order)
```

Comment:
I don't think you can update this yet. We're only populating these fields in this PR; this change is reading from them, and that's going to cause deployment issues.